client/src/components/ProfilePost/index.tsx

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import { Modal, ModalHeader } from 'reactstrap';
+import './ProfilePost.scss';
+
+interface IModalState {
+    modal: boolean;
+    flag: boolean;
+}
+
+interface IProps {
+    deletePhoto: () => void;
+    addLike: any;
+}
+
+export default class ProfilePost extends React.Component<IProps, IModalState> {
+
+    constructor(props: IProps) {
+        super(props);
+        this.state = {
+            modal: false,
+            flag: false,
+        };
+
+        this.toggle = this.toggle.bind(this);
+        this.handleCLick = this.handleCLick.bind(this);
+    }
+
+    public toggle(): any {
+        this.setState({
+            modal: !this.state.modal,
+        });
+    }
+
+    public handleCLick(): any {
+        this.setState({
+            flag: true,
+        });
+    }
+
+    public deletePhoto(): void {
+        this.props.deletePhoto();
+        this.toggle();
+    }
+
+    public addLike(): any {
+        this.props.addLike();
+        this.handleCLick();
```

Comment:
Is there only one button for click?
If no, pls give more clear name

Reply:
Okay